### ISSUE TITLE
E2e): FMC solutions similar or identical with the inverse scramble 

### ISSUE THREAD

**Issue by Laura-O** [thewca/wca-regulations#344]
After the recent discussions about Fewest Moves solutions being identical or similar with the inverse scramble, several people have suggested a clarification or modification of [E2e](https://www.worldcubeassociation.org/regulations/#E2e):

> E2e) The competitor's solution must not be directly derived from any part of the scrambling algorithm. Penalty: disqualification of the attempt (DNF).
> E2e1) The WCA Delegate may ask the competitor to explain the purpose of each move in their solution, irrespective of scrambling algorithm. If the competitor cannot give a valid explanation, the attempt is disqualified (DNF).

At the moment, I can't think of a way how we could specify which solutions can be accepted and which should be declined. For sure, it would be possible to define some rules, but these would include an advanced analysis and understanding of the solution, which means that every Delegate would need to have an advanced knowledge of FMC techniques.

However, I think one of the main problems with this regulation is the second part of E2e1), i.e. an attempt is disqualified, if the competitor cannot give a valid explanation. Many seem to interpret that this also means that a solution is valid if it can be explained, although this is a logical fallacy:
1. Every solution, which is derived from a scramble, is a DNF.
2. Every solution, which cannot be explained by the competitor, is a DNF.

If a solution is equal or similar to the inverse scramble and the competitor is able to explain his solution, only 1. (i.e. E2e)) is relevant, which means we have to decide whether a solution is derived from the scramble or not. Similarly, E2e) does not mention any intention by the competitor, so we do not have to decide whether a competitor is trustful or not or if he is capable of finding this solution.

I don't know if this is a regional phenomenon, but this is obviously another regulation which is commonly interpreted in a different way than intended. Many competitors believe that their solution is valid as long as they are able to explain it, which is definitely wrong.

An additional problem I see is the word "derived". I am not a native speaker, but "derive" sounds rather vague to me. It has also been discussed before, how FMC techniques like NISS/switching/premoves make use of the scramble, so that the solution is actually derived from the scramble when these techniques are used. What we don't want is a relation between the solution and the scramble, i.e. using the inverse scramble or parts of the scramble for the solution.

Would it improve the future application of E2e to replace "directly derived" with "related"?

> E2e) The competitor's solution must not be related to any part of the scrambling algorithm. Penalty: disqualification of the attempt (DNF).

Further sources:
- [Discussion on speedsolving.com, 2014](https://www.speedsolving.com/forum/showthread.php?13599-The-FMC-thread&p=940511&viewfull=1#post940511)
- [Another discussion, 2010](https://www.speedsolving.com/forum/showthread.php?13599-The-FMC-thread&p=546070&viewfull=1#post546070)

**Comment by Claster** [comment]
Removing scramble from FMC sheet and providing cube image only would solve these problems, though change the nature of the event drastically -_-

Does it make sense to introduce absolute thresholds? I.e. if the first or last, say, 4 moves\* are equal to last or first moves of scramble, DNF? The logic of taking 4 moves could be that if the first 3 moves of scramble or its reverse happen to solve a nice block, competitors are likely to find those even without the scramble.
- \- after removing of cancellations, rotations and probably with transposition of moves withing one axis. So, R2 R' y' F is the same as L R.

The word "derived" should be changed. If one decides to interpret this regulation as "competitor derived the solution", it becomes impossible to apply this regulation, unless a delegate can read other people's thoughts.

**Comment by Laura-O** [comment]
Yes, let's employ some little monkeys who scramble the cubes for every competitor, so they do not see the scramble. ;-)

Implementing an absolute threshold doesn't look like an applicable solution to me:
- You can easily hide moves by inserting pieces at the right position of the solution, so you could simply take the inverse scramble, do an insertion somewhere in the first 4 moves and the "problem" is solved.
- As you mentioned, you can also hide the moves at the beginning by doing stuff like cube rotations or by interchanging opposite moves. I don't think we should require that Delegates resolve those solutions.

**Comment by Randomno** [comment]
I don't really see how "related" is an improvement on "derived". Maybe the regulation could be expanded or some examples added in the Guildlines?

**Comment by Laura-O** [comment]
From my point of view "related" clarifies that no intention is needed. "Derive" is a verb, so this implies some kind of action. In contrast, "related" is about the scramble and the solution and does not suggest that a relation was intentional. This makes it way easier because this will not lead to any discussion about trust and skills.
I am not sure if this makes sense, but for me it does. :smile: 

But yes, I thought about adding examples as well. However, what are examples for solutions that we want to forbid?
Basically, the only thing I can think of is a great similarity between the inverse scramble and the solution, either by starting with the inverse scramble to build blocks or pairs, or using the complete inverse scramble and making this less obvious by changing some moves (i.e. by adding cube rotations or doing something like insertion + inverse scramble + insertion).

@SAuroux Can you think of other examples that we could mention?

**Comment by SAuroux** [comment]
I think I would list two examples:

1) Solutions starting with the inverse scrambles (>= 5 moves) [Note: with
the new scrambles generation method, 5 moves will be a good limit for this]

2) Any intentional use of the scramble sequence to derive a solution.
[Note: I would simply not be specific about this, like giving the idea to
alternate the inverse scramble in some way.]

2016-03-14 12:58 GMT+01:00 Laura Ohrndorf notifications@github.com:

**Comment by Laura-O** [comment]
Yes, sure, we shouldn't be too specific about it and avoid making the regulations some kind of cheating tutorial...
I just wanted to collect examples here in order to summarize them.

**Comment by Laura-O** [comment]
Sarah suggested this addition:
_E2e2) The solution of the competitor must not unreasonably related to the scrambling algorithm. Penalty: disqualification of the attempt (DNF), at the discretion of the WCA Delegate._

Is this reasonable if we back up the "unreasonable related" with the above mentioned examples in the Guidelines?

As an alternative we could also restructure E2e) completely:
- E2e) The solution of the competitor must not unreasonably related to the scrambling algorithm. Penalty: disqualification of the attempt (DNF), at the discretion of the WCA Delegate. This includes: 
  - E2e2) Solutions beginning with more than 4 moves of the inverse of the scrambling sequence
  - E2e3) Any intentional use of the scramble sequence to derive a solution
- E2f) The WCA Delegate may ask the competitor to explain the purpose of each move in their solution, irrespective of the scramble sequence. If the competitor cannot give a valid explanation, the attempt is disqualified (DNF).

I think I am in favor of the latter as this also clarifies that there is not necessarily a connection between the current E2e) and E2e1). The current formatting as a subregulation seems to suggest that.

In any case we should wait for the final decision how Fewest Moves scrambles will be implemented in the next TNoodle release.

**Comment by lgarron** [comment]
Almost all examples go in the Guidelines, so I think "E2e2" and "E2e3" should go there.
Also note that you're missing a word: "must not _be_ unreasonably".

The wording still doesn't sound great to me. If you asked me whether the normal T-perm and Y-perm are "reasonably related", I would say yes.
In this case, we'd be using  "_un_reasonably related" to mean basically the same thing (having related move sequences).

I personally still think the current Regulation is fine, although we could replace "scrambling algorithm" with "scrambling sequence" with consistency with 4b2.

**Comment by Laura-O** [comment]
Yes, I am not 100% satisfied with the wording as well.

Another suggestion would be to add a _"at the discretion of the WCA Delegate"_ to E2e):
- E2e) The competitor's solution must not be directly derived from any part of the scrambling sequence. Penalty: disqualification of the attempt (DNF), at the discretion of the WCA Delegate.

And then add the above mentioned examples to the Guidelines:
- E2e+) EXAMPLE Examples of solutions that should result in a disqualification of the attempt (DNF): solutions beginning with more than 4 moves of the inverse scrambling sequence or any solution intentionally derived from the scrambling sequence.

Any ideas and comments are appreciated. With the current situation in mind it would be good to have a reasonable change ASAP.

**Comment by npip99** [comment]
Not sure if I want to bring this back from the dead after 4 years, but why give them a 20-move scramble? Can't we
- (1) Generate a random 20-move scramble
- (2) Generate a solution to that scramble using CFOP
- (3) Reverse the sequence of moves generated in step (2)
- (4) Have them use a solved cube and _only_ show them the moves in step (3). _Not_ the moves in step (1). That way we give them a 50 to 60 move solution that they could have trivially produced themselves had we simply given them the scrambled cube. It's functionality equivalent to us giving them the cube with the scramble, and giving them no further useful information.

The first three steps can be generated by a computer algorithmically, with no human interaction, only a 50-60 move output.

**Comment by lgarron** [comment]
> Not sure if I want to bring this back from the dead after 4 years, but why give them a 20-move scramble? Can't we
> 
> * (1) Generate a random 20-move scramble
> * (2) Generate a solution to that scramble using CFOP
> * (3) Reverse the sequence moves generated in step (2)
> * (4) Have them use a solved cube and _only_ show them the moves in step (3). _Not_ the moves in step (1). That way we give them a 50 to 60 move solution that they could have trivially produced themselves had we simply given them the scrambled cube. It's functionality equivalent to us giving them the cube with the scramble, and giving them no further useful information.
> 
> The first three steps can be generated by a computer algorithmically, with no human interaction, only a 50-60 move output.

I could point out issues with that exact procedure, but you bring this up at a very good time:

- G0/HT/Domino reduction is becoming more common. This means that a lot more solutions are more likely to look like inverse scrambles (yes, we have padding moves, but those could also turn into premoves), and in particular *it makes it easier for someone to claim* that their solution was legit despite looking similar to an inverse scramble.
- All attempts right now are unofficial, so we have a chance to test things out when people are doing solves at home.

I don't know if it would be worth changing, but I think it could be a good time to evaluate.

**Comment by xsrvmy** [comment]
Weren't the padding moves specifically chosen so that they wouldn't make much sense from a EO perspective?

Biggest problem I see is this can severely affect time management and possibly make reading a scramble reliably too big a part of FMC. At least I think it's not as bad as removing rotations, because that only affect specific advanced techniques rather than the entire event.
